### ISSUE TITLE
TextBlocks are not read by screen readers when they are used in a XAML Islands app

### ISSUE THREAD

**Issue by DHowett** [microsoft/microsoft-ui-xaml#10869]
<html>
<body>
<div data-testid="issue-viewer-issue-container" style="box-sizing: border-box; color: rgb(240, 246, 252); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="Box-sc-62in7e-0 blqUaE" style="box-sizing: border-box; display: flex; gap: 16px;"><div data-testid="issue-body" class="Box-sc-62in7e-0 iLyfmp react-issue-body" data-hpc="true" style="box-sizing: border-box; -webkit-box-flex: 1; flex-grow: 1; order: 0; min-width: 0px;"><div class="Box-sc-62in7e-0 biaUmf" style="box-sizing: border-box; display: flex; flex-direction: row; gap: 8px;"><div class="Box-sc-62in7e-0 iOcXNN" style="box-sizing: border-box; border: 1px solid rgb(61, 68, 77); border-radius: 6px; -webkit-box-flex: 1; flex-grow: 1; width: 856px; min-width: 0px;"><div class="IssueBodyViewer-module__IssueBody--Zg6Wy" id="issue-body-viewer" data-testid="issue-body-viewer" style="box-sizing: border-box; display: flex; flex-direction: column; gap: 24px; padding: 16px;"><div data-testid="markdown-body" data-team-hovercards-enabled="true" class="markdown-body" data-turbolinks="false" style="box-sizing: border-box; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; line-height: 1.5; overflow-wrap: break-word;"><div class="Box-sc-62in7e-0 markdown-body NewMarkdownViewer-module__safe-html-box--dKCgP" style="box-sizing: border-box; font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; line-height: 1.5; overflow-wrap: break-word; max-width: 100%; margin-top: 0px !important; margin-bottom: 0px !important;"><p dir="auto" style="box-sizing: border-box; margin-top: 0px !important; margin-bottom: 16px;"><strong style="box-sizing: border-box; font-weight: 600;">Describe the bug</strong></p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;"><code class="notranslate" style="box-sizing: border-box; font-family: &quot;Monaspace Neon&quot;, ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.9px; tab-size: 4; padding: 0.2em 0.4em; margin: 0px; white-space: break-spaces; background-color: rgba(101, 108, 118, 0.2); border-radius: 6px;">TextBlock</code>s are not read by Narrator and NVDA when they are used in a XAML Islands app.</p><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;"><strong style="box-sizing: border-box; font-weight: 600;">Steps to reproduce the bug</strong><br style="box-sizing: border-box;">Verified this behavior in the following places:</p><ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px;"><li style="box-sizing: border-box;">Windows Terminal Settings UI (NavigationView scenario)</li><li style="box-sizing: border-box; margin-top: 0.25em;"><a href="https://github.com/microsoft/Xaml-Islands-Samples" style="box-sizing: border-box; background-color: rgba(0, 0, 0, 0); color: rgb(68, 147, 248); text-decoration: underline; text-underline-offset: 0.2rem;">Xaml Islands Sample App</a><ul dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 0px;"><li style="box-sizing: border-box;">"Win UI Navigation" is not read in the navigation view (even if a button is placed as the content next to the text block)</li><li style="box-sizing: border-box; margin-top: 0.25em;">(when added) a text block above the "Hello from Xaml!" text box is not read</li></ul></li></ul><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px;">I have also tried adding a hyperlink to the text blocks, and the behavior slightly changed:</p><markdown-accessiblity-table data-catalyst="" style="box-sizing: border-box; display: block;">
  | Narrator | NVDA
-- | -- | --
Normal | Reads hyperlink, but not surrounding text. | Reads text block and hyperlinked portion.
Xaml Islands | Silence. | Silence.

</markdown-accessiblity-table><p dir="auto" style="box-sizing: border-box; margin-top: 0px; margin-bottom: 0px !important;"><strong style="box-sizing: border-box; font-weight: 600;">Additional context</strong><br style="box-sizing: border-box;">Surprisingly, I've done a bit of testing with ContentDialogs too (see<span> </span><a class="issue-link js-issue-link" data-error-text="Failed to load title" data-id="639282503" data-permission-text="Title is private" data-url="https://github.com/microsoft/microsoft-ui-xaml/issues/2681" data-hovercard-type="issue" data-hovercard-url="/microsoft/microsoft-ui-xaml/issues/2681/hovercard" href="https://github.com/microsoft/microsoft-ui-xaml/issues/2681" aria-keyshortcuts="Alt+ArrowUp" style="box-sizing: border-box; background-color: rgba(0, 0, 0, 0); color: rgb(68, 147, 248); text-decoration: underline; text-underline-offset: 0.2rem;">#2681</a>). For NVDA, the text blocks are actually ready when the Content Dialog opens (except for the title). This further makes me think that the issue isn't that the automation properties aren't being exposed, but rather we need to notify the screen reader for when to read the text block.</p></div></div><div class="IssueBodyViewer-module__IssueBodyTaskList--xb3uu" style="box-sizing: border-box; align-items: center; display: flex;"><div role="toolbar" aria-label="Reactions" class="d-flex gap-1 flex-wrap" style="box-sizing: border-box; flex-wrap: wrap !important; display: flex !important; gap: 4px !important;"></div></div></div></div></div></div></div></div><div data-testid="issue-viewer-comments-container" class="react-comments-container" style="box-sizing: border-box; color: rgb(240, 246, 252); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><div class="IssueViewer-module__commentsContainer--H8wxg" style="box-sizing: border-box; margin-bottom: 16px;"><br class="Apple-interchange-newline">
</body>
</html>Describe the bug

TextBlocks are not read by Narrator and NVDA when they are used in a XAML Islands app.

Steps to reproduce the bug
Verified this behavior in the following places:

Windows Terminal Settings UI (NavigationView scenario)
[Xaml Islands Sample App](https://github.com/microsoft/Xaml-Islands-Samples)
"Win UI Navigation" is not read in the navigation view (even if a button is placed as the content next to the text block)
(when added) a text block above the "Hello from Xaml!" text box is not read
I have also tried adding a hyperlink to the text blocks, and the behavior slightly changed:

Narrator	NVDA
Normal	Reads hyperlink, but not surrounding text.	Reads text block and hyperlinked portion.
Xaml Islands	Silence.	Silence.
If you enter scan mode, you are able to navigate around the text (hyperlinked and not).

Expected behavior

At a minimum, XAML Islands should have the same behavior as a non-xaml islands app.

Ideally, for hyperlinks, all screen readers would behave like NVDA. That said, that might be an issue that's on screen readers, not XAML.

The real issue here is that these text blocks are not read automatically. It seems like they're being exposed, and we're just not notifying the screen reader that we've navigated to a portion of the screen that has a text block.

Version Info
NuGet package version:
[Microsoft.UI.Xaml 2.5.0]

Windows app type:

UWP	Win32
Windows 10 version	Saw the problem?
Insider Build (xxxxx)	
October 2020 Update (19042)	
May 2020 Update (19041)	
November 2019 Update (18363)	
May 2019 Update (18362)	
October 2018 Update (17763)	
April 2018 Update (17134)	
Fall Creators Update (16299)	
Creators Update (15063)	
Device form factor	Saw the problem?
Desktop	
Xbox	
Surface Hub	
IoT	
Additional context
Surprisingly, I've done a bit of testing with ContentDialogs too (see https://github.com/microsoft/microsoft-ui-xaml/issues/2681). For NVDA, the text blocks are actually ready when the Content Dialog opens (except for the title). This further makes me think that the issue isn't that the automation properties aren't being exposed, but rather we need to notify the screen reader for when to read the text block.

**Comment by DHowett** [comment]
I've kindly reopened a new bug to track closed issue #4399 as the team will not respond on that one.
